integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/management/ElasticsearchSchemaAttributeValidationIT.java

```diff
@@ -0,0 +1,890 @@
+
+	private static final String BACKEND_NAME = "myElasticsearchBackend";
+	private static final String INDEX_NAME = "IndexName";
+	private static final String SCHEMA_VALIDATION_CONTEXT = "schema validation";
+
+	@Rule
+	public SearchSetupHelper setupHelper = new SearchSetupHelper();
+
+	@Rule
+	public TestElasticsearchClient elasticSearchClient = new TestElasticsearchClient();
+
+	@Test
+	public void valid() {
+		elasticSearchClient.index( INDEX_NAME ).deleteAndCreate();
+		elasticSearchClient.index( INDEX_NAME ).type().putMapping(
+				"{"
+					+ "'dynamic': 'strict',"
+					+ "'properties': {"
+							+ "'myField': {"
+									+ "'type': 'integer',"
+									+ "'index': true"
+							+ "}"
+					+ "}"
+				+ "}"
+		);
+
+		validateSchemaConfig()
+				.withIndex( INDEX_NAME, ctx -> {
+							IndexSchemaElement root = ctx.getSchemaElement();
+							root.field( "myField", f -> f.asInteger() ).toReference();
+						}
+				)
+				.setup();
+	}
+
+	@Test
+	public void attribute_dynamic_missing() {
+		elasticSearchClient.index( INDEX_NAME ).deleteAndCreate();
+		elasticSearchClient.index( INDEX_NAME ).type().putMapping(
+				"{"
+					+ "'properties': {"
+							+ "'myField': {"
+									+ "'type': 'integer',"
+									+ "'index': true"
+							+ "}"
+					+ "}"
+				+ "}"
+		);
+
+		SubTest.expectException( () ->
+				validateSchemaConfig()
+						.withIndex( INDEX_NAME, ctx -> {
+									IndexSchemaElement root = ctx.getSchemaElement();
+									root.field( "myField", f -> f.asInteger() ).toReference();
+								}
+						)
+						.setup() )
+				.assertThrown()
+				.isInstanceOf( SearchException.class )
+				.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
+						.indexContext( INDEX_NAME )
+						.contextLiteral( SCHEMA_VALIDATION_CONTEXT )
+						.failure( "Invalid value for attribute 'dynamic'. Expected 'STRICT', actual is 'null'" )
+						.build() );
+	}
+
+	@Test
+	public void attribute_dynamic_invalid() {
+		elasticSearchClient.index( INDEX_NAME ).deleteAndCreate();
+		elasticSearchClient.index( INDEX_NAME ).type().putMapping(
+				"{"
+					+ "'dynamic': false,"
+					+ "'properties': {"
+							+ "'myField': {"
+									+ "'type': 'integer',"
+									+ "'index': true"
+							+ "}"
+					+ "}"
+				+ "}"
+		);
+
+		SubTest.expectException( () ->
+				validateSchemaConfig()
+						.withIndex( INDEX_NAME, ctx -> {
+									IndexSchemaElement root = ctx.getSchemaElement();
+									root.field( "myField", f -> f.asInteger() ).toReference();
+								}
+						)
+						.setup() )
+				.assertThrown()
+				.isInstanceOf( SearchException.class )
+				.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
+						.indexContext( INDEX_NAME )
+						.contextLiteral( SCHEMA_VALIDATION_CONTEXT )
+						.failure( "Invalid value for attribute 'dynamic'. Expected 'STRICT', actual is 'FALSE'" )
+						.build() );
+
+	}
+
+	@Test
+	public void attribute_properties_missing() {
```

Comment:
If we follow your line of reasoning regarding the moving of test method to a separate class, shouldn't this have remained in `ElasticsearchSchemaValidationIT`, next to `mapping_missing` and `property_fields_missing`?

Reply:
You're right. That's the reasoning.

Reply:
So `properties`, which appears at the top level unlike other attributes, is an attribute nonetheless, but `fields`, which appears just below a property like other attributes, is not (`org.hibernate.search.integrationtest.backend.elasticsearch.management.ElasticsearchSchemaValidationIT#property_fields_missing`). I don't understand, but ok, I'll merge it anyway.

Reply:
The name of the field itself is not fixed, it depends case by case. In this sense it would be not a fixed-name attribute.

Reply:
They are structure in the exact same way, except fields can only be found at the same level as other attributes, while properties can also appear at the top level:

```
"properties": {
  "propertyName": {
    "type": "object",
    "properties": {
      "propertyName1": {
        "type": "integer"
      },
      "propertyName2": {
        "type": "keyword",
        "normalizer": "foo"
      },
  }
}
```

```
"properties": {
  "propertyName": {
    "type": "keyword",
    "fields": {
      "fieldName1": {
        "type": "integer"
      },
      "fieldName2": {
        "type": "keyword",
        "normalizer": "foo"
      }
  }
}
```